specs/eip4844/p2p-interface.md

```diff
@@ -0,0 +1,264 @@
+
+### Topics and messages
+
+Topics follow the same specification as in prior upgrades.
+All topics remain stable except the beacon block topic which is updated with the modified type.
+
+The specification around the creation, validation, and dissemination of messages has not changed from the Bellatrix document unless explicitly noted here.
+
+The derivation of the `message-id` remains stable.
+
+The new topics along with the type of the `data` field of a gossipsub message are given in this table:
+
+| Name | Message Type |
+| - | - |
+| `beacon_block` | `SignedBeaconBlock` (modified) |
+| `blobs_sidecar` | `SignedBlobsSidecar` (new) |
+
+Note that the `ForkDigestValue` path segment of the topic separates the old and the new `beacon_block` topics.
+
+#### Global topics
+
+EIP4844 changes the type of the global beacon block topic and introduces a new global topic for blobs-sidecars.
+
+##### `beacon_block`
+
+The *type* of the payload of this topic changes to the (modified) `SignedBeaconBlock` found in EIP4844.
+
+In addition to the gossip validations for this topic from prior specifications,
+the following validations MUST pass before forwarding the `signed_beacon_block` on the network.
+Alias `block = signed_beacon_block.message`, `execution_payload = block.body.execution_payload`.
+- _[REJECT]_ The KZG commitments of the blobs are all correctly encoded compressed BLS G1 Points.
+  -- i.e. `all(bls.KeyValidate(commitment) for commitment in block.body.blob_kzgs)`
+- _[REJECT]_ The KZG commitments correspond to the versioned hashes in the transactions list.
+  -- i.e. `verify_kzgs_against_transactions(block.body.execution_payload.transactions, block.body.blob_kzgs)` 
+
+##### `blobs_sidecar`
+
+This topic is used to propagate data blobs included in any given beacon block.
+
+The following validations MUST pass before forwarding the `signed_blobs_sidecar` on the network;
+Alias `sidecar = signed_blobs_sidecar.message`.
+- _[IGNORE]_ the `sidecar.beacon_block_slot` is for the current slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance) -- i.e. `blobs_sidecar.beacon_block_slot == current_slot`.
+- _[REJECT]_ the `sidecar.blobs` are all well formatted, i.e. the `BLSFieldElement` in valid range (`x < BLS_MODULUS`).
+- _[REJECT]_ the beacon proposer signature, `signed_blobs_sidecar.signature`, is valid -- i.e.
+```python
+domain = get_domain(state, DOMAIN_BLOBS_SIDECAR, blobs_sidecar.beacon_block_slot / SLOTS_PER_EPOCH)
+signing_root = compute_signing_root(blobs_sidecar, domain)
+assert bls.Verify(proposer_pubkey, signing_root, signed_blob_header.signature)
+```
+  where `proposer_pubkey` is the pubkey of the beacon block proposer of `blobs_sidecar.beacon_block_slot`
+- _[IGNORE]_ The sidecar is the first sidecar with valid signature received for the `(proposer_index, sidecar.beacon_block_root)` combination,
+  where `proposer_index` is the validator index of the beacon block proposer of `blobs_sidecar.beacon_block_slot`
```

Comment:
So it's really just `beacon_block_slot, beacon_block_root` combination because we can only have one mapping for proposer_index for a given `slot` unless we know the chain. (otherwise we must assume our canonical view)